lib/cloud/imds/azure/imds.go

```diff
@@ -212,24 +212,43 @@ func (client *InstanceMetadataClient) GetHostname(ctx context.Context) (string,
 	return value, nil
 }
 
-// GetID gets the Azure resource ID of the cloud instance.
-func (client *InstanceMetadataClient) GetID(ctx context.Context) (string, error) {
-	compute := struct {
-		ResourceID string `json:"resourceId"`
-	}{}
+// InstanceInfo contains the current instance's metadata information.
+// Values obtained from
+// https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service?tabs=linux#get-user-data
```

Comment:
nit:

```suggestion
// https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service?tabs=linux#get-user-data.
```

Reply:
I don't like adding punctuation or other chars to URLs as it might break the URL.

Reply:
Most editors are clever enough to leave the final stop out of the "link" proper, but I understand.

Reply:
I think `.` is valid for fragment section, so maybe they try to be too clever 😅